DQMOffline/Trigger/plugins/HLTFiltersDQMonitor.cc

```diff
@@ -249,209 +286,256 @@ void HLTFiltersDQMonitor::analyze(const edm::Event& iEvent, const edm::EventSetu
+    edm::EDConsumerBase::Labels labels;
+    labelsForToken(triggerResultsToken_, labels);
+    edm::LogWarning("HLTFiltersDQMonitor")
+        << "invalid handle to edm::TriggerResults (InputTag: \"" << labels.module << ":" << labels.productInstance
+        << ":" << labels.process << "\") -> plugin will not fill DQM outputs for this event";
     return;
   }
 
-  // fill MonitorElement: HLT-Menu (bin: path)
-  if (meMenu_ and meMenu_->getTProfile() and meMenu_->getTProfile()->GetXaxis()) {
-    auto const& triggerNames(hltConfigProvider_.triggerNames());
-    for (auto const& iPathName : triggerNames) {
-      const uint pathIndex(hltConfigProvider_.triggerIndex(iPathName));
-      if (pathIndex >= triggerResults->size()) {
-        edm::LogError("Logic") << "[HLTFiltersDQMonitor::analyze]       "
-                               << "index associated to path \"" << iPathName << "\" (" << pathIndex
-                               << ") is inconsistent with triggerResults::size() (" << triggerResults->size()
-                               << ") -> plugin will not fill bin associated to this path in HLT-Menu MonitorElement";
-        continue;
-      }
-      auto const pathAccept(triggerResults->accept(pathIndex));
-      auto* const axis(meMenu_->getTProfile()->GetXaxis());
-      auto const ibin(axis->FindBin(iPathName.c_str()));
-      if ((0 < ibin) and (ibin <= axis->GetNbins())) {
-        meMenu_->Fill(ibin - 0.5, pathAccept);
-      }
+  // fill MonitorElement: HLT-Menu (bin: Path)
+  auto const& triggerNames = hltConfigProvider_.triggerNames();
+  for (auto const& iPathName : triggerNames) {
+    unsigned int const pathIndex = hltConfigProvider_.triggerIndex(iPathName);
+    if (pathIndex >= triggerResults->size()) {
+      edm::LogError("HLTFiltersDQMonitor")
+          << "[HLTFiltersDQMonitor::analyze]       "
+          << "index associated to Path \"" << iPathName << "\" (" << pathIndex
+          << ") is inconsistent with triggerResults::size() (" << triggerResults->size()
+          << ") -> plugin will not fill bin associated to this Path in HLT-Menu MonitorElement";
+      continue;
     }
-  }
-
-  auto const& triggerEventAOD(iEvent.getHandle(triggerSummaryTokenAOD_));
-  edm::Handle<trigger::TriggerEventWithRefs> triggerEventRAW;
 
-  bool useTriggerEventAOD(true);
-  if (not triggerEventAOD.isValid()) {
-    useTriggerEventAOD = false;
-    edm::LogInfo("Input") << "invalid handle to trigger::TriggerEvent (InputTag: \"triggerSummaryAOD\"),"
-                          << " will attempt to access trigger::TriggerEventWithRefs (InputTag: \"triggerSummaryRAW\")";
+    if (binIndexMap_.find(iPathName) == binIndexMap_.end()) {
+      throw cms::Exception("HLTFiltersDQMonitorInvalidBinLabel")
+          << "invalid key for bin-index map (name of Path bin in MonitorElement of HLT Menu): \"" << iPathName << "\"";
+    }
+    auto const ibin = binIndexMap_[iPathName];
```

Comment:
The following would be more efficient
```suggestion
    auto const foundBin = binIndexMap_.find(iPathName);
    if (foundBin == binIndexMap_.end()) {
      throw cms::Exception("HLTFiltersDQMonitorInvalidBinLabel")
          << "invalid key for bin-index map (name of Path bin in MonitorElement of HLT Menu): \"" << iPathName << "\"";
    }
    auto const ibin = *foundBin;
```
by avoiding a second search (even if it has a constant cost on average).